js/src/common/states/AlertManagerState.ts

```diff
@@ -1,30 +1,41 @@
-import Alert from '../components/Alert';
+import Mithril from 'mithril';
+import Alert, { AlertAttrs } from '../components/Alert';
+
+export interface AlertState {
+  componentClass: AlertManagerState;
+  attrs: AlertAttrs;
+  children: Mithril.Children;
+}
 
 export default class AlertManagerState {
-  constructor() {
-    this.activeAlerts = {};
-    this.alertId = 0;
-  }
+  protected activeAlerts: { [ids: string]: AlertState } = {};
+  protected alertId = 0;
 
-  getActiveAlerts() {
+  public getActiveAlerts() {
     return this.activeAlerts;
   }
 
   /**
    * Show an Alert in the alerts area.
+   *
+   * @returns The alert's ID, which can be used to dismiss the alert.
    */
-  show(arg1, arg2, arg3) {
+  public show(children: Mithril.Children): number;
+  public show(attrs: AlertAttrs, children: Mithril.Children): number;
+  public show(componentClass: Alert, attrs: AlertAttrs, children: Mithril.Children): number;
+
+  public show(arg1: Mithril.Children | AlertAttrs | Alert, arg2?: AlertAttrs | Mithril.Children, arg3?: Mithril.Children) {
```

Comment:
If I understand [the docs](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) correctly, this last signature isn't actually callable (otherwise, it would currently allow passing three arguments of type `Mithril.Children`). Therefore, we can improve readability a bit by doing `arg1: any, arg2: any, arg3: any`. (This would probably require a few typecasts below.) What do you think?

Reply:
Done, typecasts were necessary anyways